lalrpop/src/normalize/token_check/mod.rs

```diff
@@ -21,19 +21,50 @@ mod test;
 
 pub fn validate(mut grammar: Grammar) -> NormResult<Grammar> {
     let (has_enum_token, all_literals) = {
+        let opt_match_token = grammar.match_token();
```

Comment:
Side comment: Do we ever check that there is at most one `match_token`? We ought to check in prevalidate I imagine.